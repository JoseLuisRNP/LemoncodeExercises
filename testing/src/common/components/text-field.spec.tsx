import * as React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {TextField} from './text-field';

describe('Text field component specs', () => {
  it('should display a textfield with when it is feed with props', () => {
    //Arrange
    const props = {
      name: 'Textfield name',
      label: 'Textfield label',
      value: 'Textfield value',
      onChange: jest.fn(),
    }
    //Act
    const {getByTestId, asFragment} = render(<TextField {...props} />)
    const textFieldElement = getByTestId('material-textfield');
    //Assert
    expect(textFieldElement).not.toBeUndefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call onChange property when the textfield change', () => {
    //Arrange
    const props = {
      name: 'Textfield name',
      label: 'Textfield label',
      value: 'Textfield value',
      onChange: jest.fn(),
    }
    //Act
    const {getByTestId} = render(<TextField {...props} />)
    const textFieldElement = getByTestId('material-textfield');
    fireEvent.change(textFieldElement, { target: { value: "new value" } });
    //Assert
    expect(props.onChange).toHaveBeenCalled();
  });

  it('should call onBlur property when it lose the focus', () => {
    //Arrange
    const props = {
      name: 'Textfield name',
      label: 'Textfield label',
      value: 'Textfield value',
      onChange: jest.fn(),
      onBlur: jest.fn(),
    }
    //Act
    const {getByTestId} = render(<TextField {...props} />)
    const textFieldElement = getByTestId('material-textfield');
    fireEvent.blur(textFieldElement);
    //Assert
    expect(props.onBlur).toHaveBeenCalled();
  });

  it('should appear an error message if error occurs', () => {
    //Arrange
    const props = {
      name: 'Textfield name',
      label: 'Textfield label',
      value: 'Textfield value',
      onChange: jest.fn(),
      error: 'Validate error',
    }
    //Act
    const {getByText} = render(<TextField {...props}/>);
    const errorMessage = getByText(props.error);
    //Assert
    expect(errorMessage.textContent).toEqual(props.error);
  })


})