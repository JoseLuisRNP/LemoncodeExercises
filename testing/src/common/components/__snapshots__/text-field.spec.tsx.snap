// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text field component specs should display a textfield with when it is feed with props 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
    >
      Textfield label
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        data-testid="material-textfield"
        type="text"
        value="Textfield value"
      />
    </div>
  </div>
</DocumentFragment>
`;
